client/src/app/gui/rtp_streams_plot.rs

```diff
@@ -0,0 +1,179 @@
+use std::fmt::{Display, Error, Formatter};
+
+use eframe::egui;
+use eframe::egui::plot::{Plot, Points};
+use eframe::egui::{TextBuffer, Ui};
+use eframe::epaint::Color32;
+use egui::plot::Legend;
+use rtpeeker_common::packet::SessionPacket;
+use rtpeeker_common::rtp::payload_type::MediaType;
+
+use SettingsXAxis::{RawTimestamp, RtpTimestamp, SequenceNumer};
+
+use crate::streams::RefStreams;
+
+#[derive(Debug)]
+pub enum SettingsXAxis {
+    RtpTimestamp,
+    RawTimestamp,
+    SequenceNumer,
+}
+
+#[cfg_attr(feature = "serde", derive(serde::Deserialize, serde::Serialize))]
+pub struct RtpStreamsPlot {
+    streams: RefStreams,
+    settings_x_axis: SettingsXAxis,
+    requires_reset: bool,
+}
+
+impl RtpStreamsPlot {
+    pub fn new(streams: RefStreams) -> Self {
+        Self {
+            streams,
+            settings_x_axis: RtpTimestamp,
+            requires_reset: false,
+        }
+    }
+
+    pub fn ui(&mut self, ctx: &egui::Context) {
+        egui::CentralPanel::default().show(ctx, |ui| {
+            ui.vertical(|ui| {
+                let is_raw_timestamp = matches!(self.settings_x_axis, RawTimestamp);
+                let is_rtp_timestamp = matches!(self.settings_x_axis, RtpTimestamp);
+                let is_sequence_number = matches!(self.settings_x_axis, SequenceNumer);
+
+                if ui
+                    .radio(is_raw_timestamp, "X axis is packet timestamp")
+                    .clicked()
+                {
+                    self.settings_x_axis = RawTimestamp;
+                    self.requires_reset = true
+                }
+                if ui
+                    .radio(is_rtp_timestamp, "X axis is RTP timestamp")
+                    .clicked()
+                {
+                    self.settings_x_axis = RtpTimestamp;
+                    self.requires_reset = true
+                }
+                if ui
+                    .radio(is_sequence_number, "X axis is sequence number")
+                    .clicked()
+                {
+                    self.settings_x_axis = SequenceNumer;
+                    self.requires_reset = true
+                }
```

Comment:
Add `all` functions to `SettingsXAxis` that returns vector of all of the enum options and do the `ui.radio(...)` in a loop over the possible options.

Reply:
<img width="633" alt="image" src="https://github.com/LVala/rtpeeker/assets/64365037/f6765a78-cc50-4c6a-b5e8-1aabcfe2b927">
does not work, i would prefer leave it as it is